src/request/selectors/RequestDetailDataSelectors.ts

```diff
@@ -1,19 +1,73 @@
 import { IRequestState } from '../stores/IRequestState';
 
 import { createSelector } from 'reselect';
+import * as _ from 'lodash';
 
+export const getFilterState = (state: IRequestState) => state.detail.data.filters;
 export const getOperations = (state: IRequestState) => state.detail.data.operations;
-export const getSelectedIndex = (state: IRequestState) => state.detail.data.selectedIndex;
+export const getSelectedOperationIdState = (state: IRequestState) => state.detail.data.selectedOperationId;
 
 export const getTotalOperationCount = createSelector(
     getOperations,
     operations => {
         return operations.length;
     });
 
+export const getFilteredOperations = createSelector(
+    getFilterState,
+    getOperations,
+    (filterState, operations) => {
+        const hiddenDatabases = _.keys(_.omitBy(filterState, isShown => isShown));
+        
+        let filteredOperations = [];
+        
+        operations.forEach((operation, index) => {
+            if (!_.includes(hiddenDatabases, operation.database)) {
+                filteredOperations.push({
+                    ordinal: index + 1,
+                    operation: operation
+                })
+            }
+        });
+        
+        return filteredOperations;
+    });
+
+export const getSelectedOperationId = createSelector(
+    getFilteredOperations,
+    getSelectedOperationIdState,
+    (operations, selectedOperationIdState) => {       
+        return selectedOperationIdState === '' && operations.length > 0 
+            ? operations[0].operation.id 
+            : selectedOperationIdState;
+    });
+
 export const getSelectedOperation = createSelector(
+    getFilteredOperations,
+    getSelectedOperationId,
+    (operations, selectedOperationId) => {
+        // TODO: Can this (need this) be optimized by building a map of id --> operation?
+        return _.find(operations, operation => operation.operation.id === selectedOperationId);
+    });
```

Comment:
Yes, I was trying to decide whether it was worth the extra complexity in the reducer of generating both a map and sorted array of operations, just for this lookup, or whether it was simpler and performant-enough, given the likely number of data operations per request, to just scan for the selected ID.  I'm still leaning towards the latter, whereas, in the case of logging, for example, I would probably go with the two mappings (as one would likely expect far more logging statements per request than data operations).